src/main/webapp/2Dscript.js

```diff
@@ -0,0 +1,86 @@
+// These are gobal variables.
+let map;
+let pose;
+let runId;
+let dataType;
+
+fetchData();
+/**
+ * This function fetchs pose data from the RunInfo servlet,
+ * it is an asynchronous call requiring initMap() to be
+ * called after the data is fully loaded.
+ */
+function fetchData() {
+  const queryString = window.location.search;
+  const urlParams = new URLSearchParams(queryString);
+  runId = urlParams.get('id');
+  dataType = urlParams.get('dataType');
+  fetch('/getrun?id=' + runId + '&dataType=' + dataType)
+      .then((response) => response.json())
+      .then((data) => pose = data)
+      .then(() => initMap());
+}
+
+/**
+ * This function embeds the Google map interface within the html
+ * along with drawing the pose trajectory and adding a button link.
+ */
+function initMap() {
+  // This intitalizes the map.
+  map = new google.maps.Map(document.getElementById('map'), {
+    center: {lat: pose[0].lat, lng: pose[0].lng},
+    zoom: 18,
+  });
+  // This draws the pose trejectory.
+  const poseTrajectory = new google.maps.Polyline({
+    path: formatPoseData(),
+    geodesic: true,
+    strokeColor: '#FF0000',
+    strokeOpacity: 1.0,
+    strokeWeight: 2,
+  });
+  poseTrajectory.setMap(map);
+  // This adds the button called from CenterControl().
+  const centerControlDiv = document.createElement('div');
+  addCenterControl(centerControlDiv, map);
+  centerControlDiv.index = 1;
+  map.controls[google.maps.ControlPosition.TOP_CENTER].push(centerControlDiv);
+}
+
+/**
+ * This function changes the pose data to a format
+ * the Google maps javascript api can read.
+ * @return {dictionary} Returns the formatted pose data dictionary.
+ */
+function formatPoseData() {
+  const poseCoordinates = [];
+  for (let i = 0; i<pose.length; i++) {
```

Comment:
What kind of object is `pose`? I think you can use `for (point of pose)` here if it's iterable.